### DISCUSSION TITLE
Material-UI component problems in v6

### DISCUSSION THREAD

**Discussion by mortmoe** [react-hook-form/react-hook-form#2043] [Q&A]
I have some problems with Material-UI components using the inputRef that some of them expose after upgrading to V6: 
- Reset causes the values to dissapear
- When inside an array remove causes the array to dissapear from values
- Setting default true or false on Checkbox and Switch doesn't seem to work
- If I wrap them in Controller it all seems to work as expected
- Bug... or am I doing something wrong?
https://codesandbox.io/s/react-hook-form-reset-problem-9qzzj?file=/src/index.js

**Comment by bluebill1049** [comment]
For anything external component unless you are sure they do expose `ref`, otherwise i would stick with `Controller`.

**Reply by mortmoe** [reply]
So in other words... wrap all Material-UI components in Controller? Is there any other drawbacks to doing this? I mean for example would it affect performance doing 
`<Controller as={TextField} control={control} name='whatever' {...props}/>`
as opposed to
`<TextField inputRef={register} name='whatever' {...props}/>`
(Except the last one would get problems if you reset as I've shown)

It's not a big problem for me, but mui should expose the ref as "inputRef" and there are quite a few references to this in the docs. For Mui TextField it all seems to work just fine... until you try to reset the form. For Checkboxes and Switches ... it seems there are also problems with the defaults.

**Reply by bluebill1049** [reply]
Controller is just like using `ref`, it isolates the re-render within the Controller. Take a look the render counter: https://codesandbox.io/s/react-hook-form-controllerautofocus-v6-eeo66
